Marlin/src/module/planner.cpp

```diff
@@ -799,14 +797,14 @@ void Planner::calculate_trapezoid_for_block(block_t * const block, const_float_t
   uint32_t initial_rate = LROUND(block->nominal_rate * entry_factor),
            final_rate = LROUND(block->nominal_rate * exit_factor); // (steps per second)
 
-  // Legacy check against supposed timer overflow. However Stepper::calc_timer_interval() already
-  // should protect against it. But removing this code produces judder in direction-switching
-  // moves. This is because the current discrete stepping math diverges from physical motion under
-  // constant acceleration when acceleration_steps_per_s2 is large compared to initial/final_rate.
+  // Limit minimal step rate (Otherwise the timer will overflow.)
   NOLESS(initial_rate, uint32_t(MINIMAL_STEP_RATE));  // Enforce the minimum speed
   NOLESS(final_rate, uint32_t(MINIMAL_STEP_RATE));
-  NOMORE(initial_rate, block->nominal_rate);          // NOTE: The nominal rate may be less than MINIMAL_STEP_RATE!
-  NOMORE(final_rate, block->nominal_rate);
+  NOLESS(block->nominal_rate, MINIMAL_STEP_RATE);
+
+  // Assume these are handled elsewhere. Retained for debugging.
+  //NOMORE(initial_rate, block->nominal_rate);        // NOTE: The nominal rate may be less than MINIMAL_STEP_RATE!
+  //NOMORE(final_rate, block->nominal_rate);
```

Comment:
These can be removed.

As a side note, I realized that with `CEIL` in was possible that `initial/final_rate` ended rounded up above `nominal_rate`. However, now that we use `LROUND` that shouldn't be possible.